### PR TITLE
Add quiz progress sign-in / bugfix

### PR DESCRIPTION
**Pull request by lrboyer**
Goal: Add quiz progress table on sign in page so users can see their quiz progress after successful sign in (Makerspace staff asked for this new feature)

Solution:
Add new component to sign in page that will fetch quiz progress and display it.

How:
Adds new query parameter on sign in so new component can grab the username 

BUGFIX:
Local hosting was not working due to "global" not being defined, fixed it in vite.config

### CODE REVIEW

**Comment by ophilli** [comment]
Can you share a screenshot of how this quiz info renders?

**Comment by ophilli** [comment]
`npm run build` is failing in the `Visitor Frontend Linting` step & when I attempt to build it locally.

File: site/visitor-console/vite.config.ts [hunk 1]
```diff
@@ -5,4 +5,7 @@ import eslintPlugin from "vite-plugin-eslint";
 // https://vitejs.dev/config/
 export default defineConfig({
   plugins: [eslintPlugin({}), react()],
+  define: {
+    global: {},
+  },
```
Comment (ophilli) [inline site/visitor-console/vite.config.ts]: This is causing `npm run build` to fail. If I comment it out then the build succeeds again, but the page doesn't render anything if I load it via `npm run start`.

**Review by ophilli — CHANGES_REQUESTED** [review]
I believe the code pipeline needs `npm run build` so merging this as is will break the pipeline.

**Comment by lrboyer** [comment]
> Can you share a screenshot of how this quiz info renders?

File: site/visitor-console/vite.config.ts (hunk 1 above)
Comment (lrboyer) [reply site/visitor-console/vite.config.ts]: Weird behavior, I forgot to do the npm run build step before making a pr. Maybe put the define global in the readme for local testing?

**Comment by ophilli** [comment]
That screenshot link you posted seems to just point to https://github.com/clemsonMakerspace/unified-makerspace/pull/153 and not to a screenshot. Can you try again?

File: site/visitor-console/vite.config.ts (hunk 1 above)
Comment (ophilli) [reply site/visitor-console/vite.config.ts]: Can you try https://github.com/vitejs/vite/discussions/5912#discussioncomment-5569850 instead? It worked on my machine for both `npm run start` and `npm run build`.

**Comment by lrboyer** [comment]
MakerspaceSigninPrototype

File: site/visitor-console/src/pages/SignInSuccess.tsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 const SignInSuccess = () => {
   const navigate = useNavigate();
   let [searchParams] = useSearchParams();
   const next_page = (searchParams.get("next") as string) || "/";
 
   const renderer = () => (
-    <PageCard title="Sign-In Sucessful">
+    <PageCard title="Sign-In Successful">
+      <h1>Your Safety Quiz Progress</h1>
+      <QuizProgress />
```
Comment (ophilli) [inline site/visitor-console/src/pages/SignInSuccess.tsx:14]: This automatically times out and goes back to the login page after 10 seconds?

**Review by ophilli — CHANGES_REQUESTED** [review]
Can you please squash your 4 commits into one single commit?

File: site/visitor-console/src/pages/SignInSuccess.tsx (hunk 2 above)
Comment (lrboyer) [reply site/visitor-console/src/pages/SignInSuccess.tsx:14]: Yes, after a user signs in, it goes back to the login page so another user can sign in after them.

**Comment by lrboyer** [comment]
> Can you please squash your 4 commits into one single commit?

I think I did that correctly

**Comment by ophilli** [comment]
Waiting on [@plantstevie ](https://github.com/plantstevie) to review / approve the UX changes.

**Review by plantstevie — APPROVED** [review]

**Review by plantstevie — APPROVED** [review]

**Review by ophilli — APPROVED** [review]

**Pull request merged**